### PR TITLE
made large file behavior smarter

### PR DESCRIPTION
**Pull request by stevekuznetsov**
@bparees This truncates large logs to 20M.

### CODE REVIEW

File: hack/util.sh [hunk 1]
```diff
@@ -485,12 +485,21 @@ function dump_container_logs()
...
+}
+
+# truncate_large_logs truncates large logs so we only download the last 20MB
+function truncate_large_logs() {
 	# Clean up large log files so they don't end up on jenkins
-	find "${ARTIFACT_DIR}" "${LOG_DIR}" -type f -name '*.log' \( -empty -or -size +20M \) -delete
+	local large_files=$(find "${ARTIFACT_DIR}" "${LOG_DIR}" -type f -name '*.log' \( -size +20M \))
+	for file in "${large_files}"; do
+		cp "${file}" "${file}.tmp"
+		tail -c 20M "${file}.tmp" > "${file}"
```
Comment (bparees) [inline hack/util.sh]: tack a header on the log file indicating it was truncated so people aren't baffled later as to why it's cut off.

**Comment by bparees** [comment]
one nit and lgtm

**Comment by stevekuznetsov** [comment]
Is 20M a technical limitation? Do we need the file to be no more than 20M,
truncating it to 20M - the header?

**Comment by bparees** [comment]
no, it was an arbitrary choice.

On Fri, Jan 8, 2016 at 1:05 PM, Steve Kuznetsov <notifications@github.com>
wrote:

> Is 20M a technical limitation? Do we need the file to be no more than 20M,
> truncating it to 20M - the header?
> On Jan 8, 2016 10:57 AM, "Ben Parees" <notifications@github.com> wrote:
>
> > one nit and lgtm
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <https://github.com/openshift/origin/pull/6595#issuecomment-170074418>.
>
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/6595#issuecomment-170076189>.
>



-- 
Ben Parees | OpenShift

**Comment by stevekuznetsov** [comment]
@bparees added a header

**Comment by bparees** [comment]
lgtm

**Comment by deads2k** [comment]
[merge]

**Pull request merged**